### PR TITLE
feat(Toolbar&OverflowMenu): update docs to use OverflowMenu components

### LINKED ISSUE #4420: Toolbar - update examples to use the actual overflow menu component
the examples currently just use stand in text 'overflow menu' rather than demonstrating how to use it. Many consumers are asking about hiding buttons and other things at different break points without realizing that the capability exists with an Overflow Menu.

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #4420 and adds the OverflowMenu to the following components: 

- Stacked Toolbar
- Card View
- Master Card View Detail Demo
- Table Column Management Demo

### CODE REVIEW

**Comment by jenny-s51** [comment]
@nicolethoen thank you for your feedback! I definitely agree with what you're saying and it would be great to get @mattnolting 's thoughts on this -- I'm also not completely certain what the "Overflow Menu" text is meant to do there ... Does the it that the whole toolbar should be an overflow menu? Or should it be something else with other contents like you're suggesting? Happy to make whatever changes make the most sense here.

**Comment by mattnolting** [comment]
`Overflow menu` is a placeholder from a long time ago that was intended to be replaced when the `overflow menu` was ready for use. Anytime you see `Overflow menu`, it's fairly safe to assume that it should be replaced by this (unless otherwise prescribed by design):

Screen Shot 2020-06-30 at 2 40 35 PM

So the `toggle-group` handles the filters, and the `overflow-menu` s handle the collapsing buttons. In this example, https://www.patternfly.org/v4/documentation/core/components/toolbar#expanded-elements, `pf-m-icon-button-group` signifies that the icon buttons aren't intended to be added to an `overflow menu`, but rather they stay static while `toggle-group` and the last toolbar item (`overflow menu`) collapse around it.

**Comment by mattnolting** [comment]
Review comments: https://marvelapp.com/prototype/8hg81if/screen/70914078

File: packages/react-core/src/demos/CardView/examples/CardView.md [hunk 1]
```diff
@@ -549,39 +560,43 @@ class CardViewBasic extends React.Component {
...
-      </DropdownItem>
+      </OverflowMenuDropdownItem>
     ];
 
     const toolbarItems = (
       <React.Fragment>
         <ToolbarItem variant="bulk-select">{this.buildSelectDropdown()}</ToolbarItem>
-        <ToolbarItem>{this.buildFilterDropdown()}</ToolbarItem>
+        <ToolbarItem toggleIcon={<FilterIcon />} breakpoint="xl">{this.buildFilterDropdown()}</ToolbarItem>
         <ToolbarItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/CardView/examples/CardView.md:584]:
```suggestion
        <ToolbarItem variant="overflow-menu">
```

File: packages/react-core/src/demos/CardView/examples/CardView.md [hunk 2]
```diff
@@ -549,39 +560,43 @@ class CardViewBasic extends React.Component {
...
+        <OverflowMenu breakpoint="xl">
```
Comment (mattnolting) [inline packages/react-core/src/demos/CardView/examples/CardView.md:585]:
```suggestion
        <OverflowMenu breakpoint="md">
```

File: packages/react-core/src/demos/CardView/examples/CardView.md [hunk 3]
```diff
@@ -549,39 +560,43 @@ class CardViewBasic extends React.Component {
...
+        <OverflowMenuItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/CardView/examples/CardView.md:586]: Looks like indentation is off in this block

File: packages/react-core/src/demos/MasterDetailDemo.md [hunk 4]
```diff
@@ -1571,19 +1579,23 @@ class MasterDetailCardView extends React.Component {
...
-          <Dropdown
-            onSelect={this.onToolbarKebabDropdownSelect}
-            toggle={<KebabToggle onToggle={this.onToolbarKebabDropdownToggle} id="card-view-data-toolbar-dropdown" />}
-            isOpen={isLowerToolbarKebabDropdownOpen}
-            isPlain
-            dropdownItems={toolbarKebabDropdownItems}
-          />
-        </ToolbarItem>
+        <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">{this.buildFilterDropdown()}</ToolbarToggleGroup>
+          <ToolbarItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/MasterDetailDemo.md:1583]:
```suggestion
          <ToolbarItem variant="overflow-menu">
```

File: packages/react-core/src/demos/MasterDetailDemo.md [hunk 5]
```diff
@@ -1571,19 +1579,23 @@ class MasterDetailCardView extends React.Component {
...
+            <OverflowMenu breakpoint="xl">
+              <OverflowMenuItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/MasterDetailDemo.md:1585]:
```suggestion
              <OverflowMenuItem isPersistent>
```

File: packages/react-core/src/demos/MasterDetailDemo.md [hunk 6]
```diff
@@ -1571,19 +1579,23 @@ class MasterDetailCardView extends React.Component {
...
-        <ToolbarItem>
-          <Dropdown
-            onSelect={this.onToolbarKebabDropdownSelect}
-            toggle={<KebabToggle onToggle={this.onToolbarKebabDropdownToggle} id="card-view-data-toolbar-dropdown" />}
-            isOpen={isLowerToolbarKebabDropdownOpen}
-            isPlain
-            dropdownItems={toolbarKebabDropdownItems}
-          />
-        </ToolbarItem>
+        <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">{this.buildFilterDropdown()}</ToolbarToggleGroup>
```
Comment (mattnolting) [inline packages/react-core/src/demos/MasterDetailDemo.md:1582]: In this example, the select doesn't need to be `hidden/expanded`, so you don't need to use a `toggle-group` to hide filters.

File: packages/react-core/src/demos/TableColumnManagementDemo.md [hunk 7]
```diff
@@ -544,24 +552,35 @@ class ColumnManagementAction extends React.Component {
...
+        <OverflowMenu breakpoint="xl">
+          <OverflowMenuItem>
+            <Button variant="primary">Action</Button>
+          </OverflowMenuItem>
+          <OverflowMenuItem>
+            <Button variant="link" onClick={this.handleModalToggle}>Manage columns</Button>
+          </OverflowMenuItem>
+        </OverflowMenu>
+      </ToolbarItem>
+            <ToolbarItem variant="pagination" alignment={{ default: 'alignRight' }}>
```
Comment (mattnolting) [inline packages/react-core/src/demos/TableColumnManagementDemo.md:565]:
```suggestion
            <ToolbarItem variant="pagination">
```

File: packages/react-core/src/demos/TableColumnManagementDemo.md [hunk 8]
```diff
@@ -544,24 +552,35 @@ class ColumnManagementAction extends React.Component {
           }
         />
       </ToolbarItem>
-      <ToolbarGroup variant="button-group">
-      <ToolbarItem><Button variant="primary">Action</Button></ToolbarItem>
-      <ToolbarItem><Button variant="link" onClick={this.handleModalToggle}>Manage columns</Button></ToolbarItem>
-      </ToolbarGroup>
+      <ToolbarItem>
+        <OverflowMenu breakpoint="xl">
```
Comment (mattnolting) [inline packages/react-core/src/demos/TableColumnManagementDemo.md:556]:
```suggestion
        <OverflowMenu breakpoint="md">
```

File: packages/react-core/src/demos/TableColumnManagementDemo.md [hunk 9]
```diff
@@ -522,15 +528,17 @@ class ColumnManagementAction extends React.Component {
...
-      <ToolbarItem>
-        <Select
-          id="page-layout-table-column-management-action-toolbar-top-select-checkbox-toggle"
-          variant={SelectVariant.single}
-          aria-label="Select Input"
-          aria-labelledby="page-layout-table-column-management-action-toolbar-top-select-checkbox-label page-layout-table-column-management-action-toolbar-top-select-checkbox-toggle"
-          placeholderText={<><FilterIcon /> Name</>}
-        />
-      </ToolbarItem>
+      <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
```
Comment (mattnolting) [inline packages/react-core/src/demos/TableColumnManagementDemo.md:531]: This example doesn't need a `toggle-group` either

File: packages/react-core/src/demos/CardView/examples/CardView.md (hunk 3 above)
Comment (jenny-s51) [reply packages/react-core/src/demos/CardView/examples/CardView.md:586]: will fix

**Pull request merged**